packages/backend/src/core/discovery/DiscoveryRunner.ts

```diff
@@ -71,6 +72,32 @@ export class DiscoveryRunner {
     )
   }
 
+  private async discoverWithRetry(
+    config: DiscoveryConfig,
+    blockNumber: number,
+  ) {
+    let discovery: DiscoveryOutput | undefined = undefined
+
+    try {
+      discovery = await this.discover(config, blockNumber)
+    } catch (error) {
+      if (error instanceof Error) {
+        if (error.message.includes('network timeout')) {
+          try {
+            discovery = await this.discover(config, blockNumber)
+          } catch {
+            throw error
+          }
+        } else {
+          throw error
+        }
+      } else {
+        throw error
+      }
```

Comment:
Use the negation and early `return` (early `throw` in this case) to avoid nesting